test-project/build.gradle

```diff
@@ -70,7 +70,14 @@ subprojects {
     apply plugin: 'application'
 
     ext.substrateVersion = "0.0.3"
-
+    def propFile = file('../../gradle.properties')
```

Comment:
`gradle.properties` is located one directory up, not two? It might be safer to use `file("${project.rootDir}/../gradle.properties")` instead.

Reply:
Note that we are running from the subproject tests, like `helloWorld`, so that's why we need to go two directories up.

So that doesn't work, `project.rootDir` will give the subproject file:
`test-project/helloWorld/gradle.properties`

Reply:
`project.rootDir` will give the project directory of the root project, as [stated in the docs](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:rootDir). So it should work.